hpOneView/resources/settings/firmware_drivers.py

```diff
@@ -88,7 +88,28 @@ def get_by(self, field, value):
             list: list of firmware baseline resources
 
         """
-        return self._client.get_by(field, value)
+        firmwares = self.get_all()
+        matches = []
+        for item in firmwares:
+            if item.get(field) == value:
+                matches.append(item)
+        return matches
+
+    def get_by_file_name(self, file_name):
+        """
+        Gets a firmware resource with match the file name.
+
+        Args:
+            file_name: file name to filter (without path)
+        Returns:
+            dict: firmware baseline resource
+
+        """
+        firmwares = self.get_all()
+        for firmware in firmwares:
+            if firmware['fwComponents'][0]['fileName'] == file_name:
```

Comment:
firmware['fwComponents'][1] for example, will never have a valid fileName to compare?